test/unit/calculateOptimalPoolAllocation.js

```diff
@@ -436,41 +437,79 @@ describe('calculateOptimalPoolAllocation', function () {
     expect(allocations.length).to.be.equal(0);
   });
 
+  it('computes optimal pool allocation across 4 pools', () => {
+    const pool1 = {
+      basePrice: BigNumber.from('300'),
+      initialCapacityUsed: BigNumber.from('0'),
+      totalCapacity: BigNumber.from('7228110000000000000000'),
+    };
+    const pool2 = {
+      basePrice: BigNumber.from('600'),
+      initialCapacityUsed: BigNumber.from('0'),
+      totalCapacity: BigNumber.from('8390570000000000000000'),
+    };
+    const pool3 = {
+      basePrice: BigNumber.from('103'),
+      initialCapacityUsed: BigNumber.from('2741190000000000000000'),
+      totalCapacity: BigNumber.from('68382300000000000000000'),
+    };
+    const pool4 = {
+      basePrice: BigNumber.from('103'),
+      initialCapacityUsed: BigNumber.from('353900000000000000000'),
+      totalCapacity: BigNumber.from('23242550000000000000000'),
+    };
+
+    let i = INITIAL_POOL_INDEX;
+    const pools = [pool1, pool2, pool3, pool4];
+    pools.forEach(pool => {
+      pool.poolId = i++;
+    });
+
+    const amount = BigNumber.from('88600380000000000000000');
+    const minUnitSize = BigNumber.from('188880198128988825736');
+    const optimalAllocations = calculateOptimalPoolAllocation(amount, pools, minUnitSize);
+
+    expect(optimalAllocations[pool1.poolId].toString()).to.be.equal('6202026600000000000000');
+    expect(optimalAllocations[pool2.poolId].toString()).to.be.equal('886003800000000000000');
+    expect(optimalAllocations[pool3.poolId].toString()).to.be.equal('60248258400000000000000');
+    expect(optimalAllocations[pool4.poolId].toString()).to.be.equal('21264091200000000000000');
+  });
+
   it('returns the same results as the brute force optimization for 6 pools with surge pricing', () => {
     const pool1 = {
       basePrice: BigNumber.from('210'),
-      initialCapacityUsed: parseEther('8990'),
-      totalCapacity: parseEther('10000'),
+      initialCapacityUsed: parseEther('899'),
+      totalCapacity: parseEther('1000'),
     };
 
     const pool2 = {
       basePrice: BigNumber.from('200'),
-      initialCapacityUsed: parseEther('8990'),
-      totalCapacity: parseEther('10000'),
+      initialCapacityUsed: parseEther('899'),
```

Comment:
i reduced size here because it's compared against the brute force algorithm and with smaller unit size brute force spins for a very long time

test case remains the same just reduced everything by 1 order of mag